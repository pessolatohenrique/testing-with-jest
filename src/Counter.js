import React, { Component } from 'react';

class Counter extends Component {
    render() {
        return (
            <div className="counter">
                <h2>VocÃª tem { this.props.count * 3 } counters</h2>
            </div>
        )
    }
}

export default Counter;