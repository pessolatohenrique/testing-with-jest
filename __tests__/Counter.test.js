import React from 'react';
import { shallow } from 'enzyme';
import Counter from '../src/Counter';

describe('Testing Counter component', () => {
    it('should render correctyl', () => {
        const wrapper = shallow(
            <Counter count={5} />
        );

        expect(wrapper.debug()).toMatchSnapshot();
        wrapper.setProps({ count: 10 });
        expect(wrapper.debug()).toMatchSnapshot();
    })
});