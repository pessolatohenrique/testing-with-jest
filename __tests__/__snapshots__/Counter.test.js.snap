// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Counter component should render correctyl 1`] = `
"<div className=\\"counter\\">
  <h2>
    Você tem 
    15
     counters
  </h2>
</div>"
`;

exports[`Testing Counter component should render correctyl 2`] = `
"<div className=\\"counter\\">
  <h2>
    Você tem 
    30
     counters
  </h2>
</div>"
`;
